tasks/check-data-keys/check-data-keys.yaml

```diff
@@ -55,21 +62,32 @@ spec:
             exit 1
         fi
 
-        RC=0
+        if ! curl -s --fail-with-body "$(params.schema)" -o /tmp/schema.json ; then
+            echo "Failed to fetch schema from $(params.schema)"
+            cat /tmp/schema.json
+            exit 1
+        fi
 
+        RC=0
         for SYSTEM in "$@" ; do
-            if [[ $(jq -r ".$SYSTEM" <<< "$KEYS_JSON") == null ]] ; then
-                echo "Unsupported system value: $SYSTEM"
-                RC=1
-            fi
-
-            echo "Checking all required keys exist for: $SYSTEM"
-            for KEY in $(jq -r ".${SYSTEM}[]" <<< "$KEYS_JSON") ; do
-                if [[ $(jq ".$SYSTEM.$KEY" "$(workspaces.data.path)/$(params.dataPath)") == "null" ]] ; then
-                    echo "Missing required $SYSTEM key: $KEY"
-                    RC=1
-                fi
-            done
+          if [[ $(jq -r ".$SYSTEM" <<< "$KEYS_JSON") == null ]] ; then
+              echo "Unsupported system value: $SYSTEM"
+              RC=1
+        fi
         done
 
-        exit $RC
+        if [ $RC -ne 0 ] ; then
+            exit 1
+        fi
+      
+        for system in "$@"; do
+          REQUIRED_KEYS=$(echo "$KEYS_JSON" | jq -c --arg system "$system" '.[$system]')
+
+          jq --arg system "$system" --argjson requiredKeys "$REQUIRED_KEYS" \
```

Comment:
But now you don't even check that `$system` is even in the data json, right? Like if it doesn't have `releaseNotes` at all - because that one is still optional.

Reply:
Went back to the old way